robotidy/transformers.py

```diff
@@ -73,3 +84,140 @@ def visit_KeywordSection(self, node):  # noqa
 
     def visit_CommentSection(self, node):  # noqa
         return self.check_if_empty(node)
+
+
+def insert_separators(indent, tokens, formatting_config):
+    yield Token(Token.SEPARATOR, indent + formatting_config.space_count * ' ')
+    for token in tokens[:-1]:
+        yield token
+        yield Token(Token.SEPARATOR, formatting_config.space_count * ' ')
+    yield tokens[-1]
+    yield Token(Token.EOL, formatting_config.line_sep)
+
+
+@transformer
+class ReplaceRunKeywordIf(ModelTransformer):
+    """
+    Replace `Run Keyword If` keyword calls with IF END blocks.
+    Supports global formatting params: `--lineseparator`, `--spacecount`.
+
+    Following code::
+
+        Run Keyword If  ${condition}
+        ...  Keyword  ${arg}
+        ...  ELSE IF  ${condition2}  Keyword2
+        ...  ELSE  Keyword3
+
+    Will be transformed to::
+
+        IF    ${condition}
+            Keyword    ${arg}
+        ELSE IF    ${condition2}
+            Keyword2
+        ELSE
+            Keyword3
+        END
+
+    Any return value will be applied to every ELSE/ELSE IF branch::
+
+        ${var}  Run Keyword If  ${condition}  Keyword  ELSE  Keyword2
+
+    Output::
+
+        IF    ${condition}
+            ${var}    Keyword
+        ELSE
+            ${var}    Keyword2
+        END
+
+    Run Keywords inside Run Keyword If will be splitted into separate keywords::
+
+       Run Keyword If  ${condition}  Run Keywords  Keyword  ${arg}  AND  Keyword2
+
+    Output::
+
+        IF    ${condition}
+            Keyword    ${arg}
+            Keyword2
+        END
+
+    """
+    def visit_KeywordCall(self, node):  # noqa
+        if not node.keyword:
+            return node
+        if normalize_name(node.keyword) == 'runkeywordif':
+            return self.create_branched(node)
+        return node
+
+    def create_branched(self, node):
+        separator = node.tokens[0]
+        assign = node.get_tokens(Token.ASSIGN)
+        raw_args = node.get_tokens(Token.ARGUMENT)
+        if len(raw_args) < 2:
+            return node
+        end = End([
+            separator,
+            Token(Token.END, 'END'),
+            Token(Token.EOL, self.formatting_config.line_sep)
+        ])
+        prev_if = None
+        for branch in reversed(list(self.split_args_on_delimeters(raw_args, ('ELSE', 'ELSE IF')))):
+            if branch[0].value == 'ELSE':
+                header = ElseHeader([
+                    separator,
+                    Token('ELSE', 'ELSE'),
```

Comment:
1. Should use `Token.ELSE` instead of `'ELSE'` as the token type. Values may change (e.g. to enums) but the constant name won't.
2. With the just committed RF code, there's no need to pass `'ELSE'` as the value anymore. Just `Token(Token.ELSE)` is enough.

Reply:
Thanks for the change in RF code - I've applied 1) and 2) suggestions to the code in latest commit.